### PR TITLE
Self help polish

### PR DESCRIPTION
**Pull request by mcgallan**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
I added some comments in the api-view: https://apiview.dev/Assemblies/Review/488037ee7b3a40faa16a4fcd8dced814

File: sdk/selfhelp/Azure.ResourceManager.SelfHelp/api/Azure.ResourceManager.SelfHelp.netstandard2.0.cs [hunk 1]
```diff
@@ -1,167 +1,167 @@
...
+        public static Azure.ResourceManager.SelfHelp.Models.SelfHelpSolutionMetadata SelfHelpSolutionMetadata(Azure.Core.ResourceIdentifier id = null, string name = null, Azure.Core.ResourceType resourceType = default(Azure.Core.ResourceType), Azure.ResourceManager.Models.SystemData systemData = null, string solutionId = null, string solutionType = null, string solutionDescription = null, System.Collections.Generic.IEnumerable<System.Collections.Generic.IList<string>> requiredParameterSets = null) { throw null; }
     }
-    public partial class CheckNameAvailabilityContent
+    public partial class SelfHelpCheckNameAvailabilityContent
     {
-        public CheckNameAvailabilityContent() { }
-        public string Name { get { throw null; } set { } }
+        public SelfHelpCheckNameAvailabilityContent() { }
+        public string ResourceName { get { throw null; } set { } }
         public string ResourceType { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/selfhelp/Azure.ResourceManager.SelfHelp/api/Azure.ResourceManager.SelfHelp.netstandard2.0.cs:61]: Enable the debug mode to get the sterilized name then change the `ResourceType`
mgmt-debug: 
  show-serialized-names: true

Comment (mcgallan) [reply sdk/selfhelp/Azure.ResourceManager.SelfHelp/api/Azure.ResourceManager.SelfHelp.netstandard2.0.cs]: fixed

**Comment by ArthurMa1978** [comment]
@mcgallan please just contribute to https://github.com/Azure/azure-sdk-for-net/pull/36623

**Pull request closed without merging**